### PR TITLE
[master] added Lesson 16

### PR DESCRIPTION
**Pull request by spanteleyko**

### CODE REVIEW

File: Lesson16/src/com/gmail/panser81/lesson16/helpers/Helper.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.gmail.panser81.lesson16.helpers;
+
+import java.util.Random;
+
+public class Helper {
+    private static Random RAND = new Random();
```
Comment (aperavoznikau-idt) [inline Lesson16/src/com/gmail/panser81/lesson16/helpers/Helper.java:6]: please use final modificator

File: Lesson16/src/com/gmail/panser81/lesson16/services/Info.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import java.time.format.DateTimeFormatter;
+
+public class Info {
+    private static final long PAUSE_TIME_MILLISECONDS = 5000;
+    private static final String DATETIME_PATTERN = "yyyy/MM/dd HH:mm:ss";
+
+    public synchronized void printInfo(String name) {
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern(DATETIME_PATTERN);
+        LocalDateTime now = LocalDateTime.now();
+        String currentDateTimeString = dateTimeFormatter.format(now);
```
Comment (aperavoznikau-idt) [inline Lesson16/src/com/gmail/panser81/lesson16/services/Info.java:13]: Better to create method for this in helper class

File: Lesson16/src/com/gmail/panser81/lesson16/services/Info.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+        System.out.format("%s - %s\n", name, currentDateTimeString);
+        try {
+            Thread.sleep(PAUSE_TIME_MILLISECONDS);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
```
Comment (aperavoznikau-idt) [inline Lesson16/src/com/gmail/panser81/lesson16/services/Info.java:19]: general approach - print message + print stacktrace

File: Lesson16/src/com/gmail/panser81/lesson16/services/impl/ArrayLessonThreadImpl.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import static com.gmail.panser81.lesson16.helpers.Helper.getRandomNumber;
+
+public class ArrayLessonThreadImpl extends Thread {
+    private static final int ARRAY_SIZE = 10;
+    private static final int MIN_VALUE = 1;
+    private static final int MAX_VALUE = 100;
```
Comment (aperavoznikau-idt) [inline Lesson16/src/com/gmail/panser81/lesson16/services/impl/ArrayLessonThreadImpl.java:11]: Better to use separate class for constants if we have a lot of objects which we create from this class

File: Lesson16/src/com/gmail/panser81/lesson16/services/impl/CreateFileLessonServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package com.gmail.panser81.lesson16.services.impl;
+
+import com.gmail.panser81.lesson16.services.*;
+
+public class CreateFileLessonServiceImpl implements LessonService {
+    private static final int THREADS_COUNT = 5;
+
+    @Override
+    public void execute() throws Exception {
+        Thread[] threads = new Thread[THREADS_COUNT];
```
Comment (aperavoznikau-idt) [inline Lesson16/src/com/gmail/panser81/lesson16/services/impl/CreateFileLessonServiceImpl.java:10]: For Java, we prefer to use collection, not array for each case

File: Lesson16/src/com/gmail/panser81/lesson16/services/impl/ArrayLessonThreadImpl.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package com.gmail.panser81.lesson16.services.impl;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import static com.gmail.panser81.lesson16.helpers.Helper.getRandomNumber;
+
+public class ArrayLessonThreadImpl extends Thread {
```
Comment (aperavoznikau-idt) [inline Lesson16/src/com/gmail/panser81/lesson16/services/impl/ArrayLessonThreadImpl.java:8]: Thread is not a Service class. Naming should be appropriate - ArrayLessonThread is enough

**Review by aperavoznikau-idt — CHANGES_REQUESTED** [review]

File: Lesson16/src/com/gmail/panser81/lesson16/helpers/Helper.java (hunk 1 above)
Comment (spanteleyko) [reply Lesson16/src/com/gmail/panser81/lesson16/helpers/Helper.java]: added

File: Lesson16/src/com/gmail/panser81/lesson16/services/Info.java (hunk 2 above)
Comment (spanteleyko) [reply Lesson16/src/com/gmail/panser81/lesson16/services/Info.java]: created helper method

File: Lesson16/src/com/gmail/panser81/lesson16/services/Info.java (hunk 3 above)
Comment (spanteleyko) [reply Lesson16/src/com/gmail/panser81/lesson16/services/Info.java:16]: fixed

File: Lesson16/src/com/gmail/panser81/lesson16/services/impl/ArrayLessonThreadImpl.java (hunk 4 above)
Comment (spanteleyko) [reply Lesson16/src/com/gmail/panser81/lesson16/services/impl/ArrayLessonThreadImpl.java]: created a static class

File: Lesson16/src/com/gmail/panser81/lesson16/services/impl/CreateFileLessonServiceImpl.java (hunk 5 above)
Comment (spanteleyko) [reply Lesson16/src/com/gmail/panser81/lesson16/services/impl/CreateFileLessonServiceImpl.java]: used a List

File: Lesson16/src/com/gmail/panser81/lesson16/services/impl/ArrayLessonThreadImpl.java (hunk 6 above)
Comment (spanteleyko) [reply Lesson16/src/com/gmail/panser81/lesson16/services/impl/ArrayLessonThreadImpl.java]: renamed

**Review by aperavoznikau-idt — APPROVED** [review]

**Pull request merged**